src/compute/leaf.rs

```diff
@@ -1,55 +1,52 @@
 //! Computes size using styles and measure functions
 
-use crate::compute::LayoutAlgorithm;
 use crate::geometry::{Point, Size};
 use crate::layout::{SizeAndBaselines, SizingMode};
 use crate::math::MaybeMath;
+use crate::node::Taffy;
 use crate::resolve::{MaybeResolve, ResolveOrZero};
 use crate::style::AvailableSpace;
 use crate::sys::f32_max;
-use crate::tree::{LayoutTree, NodeId};
+use crate::tree::NodeId;
 
 #[cfg(feature = "debug")]
 use crate::debug::NODE_LOGGER;
 
-/// The public interface to Taffy's leaf node algorithm implementation
-pub(crate) struct LeafAlgorithm;
-impl LayoutAlgorithm for LeafAlgorithm {
-    const NAME: &'static str = "LEAF";
-
-    fn perform_layout(
-        tree: &mut impl LayoutTree,
-        node: NodeId,
-        known_dimensions: Size<Option<f32>>,
-        parent_size: Size<Option<f32>>,
-        available_space: Size<AvailableSpace>,
-        sizing_mode: SizingMode,
-    ) -> SizeAndBaselines {
-        compute(tree, node, known_dimensions, parent_size, available_space, sizing_mode)
-    }
+/// Perform full layout on a leaf node
+pub(crate) fn perform_layout(
+    tree: &mut Taffy,
+    node: NodeId,
+    known_dimensions: Size<Option<f32>>,
+    parent_size: Size<Option<f32>>,
+    available_space: Size<AvailableSpace>,
+    sizing_mode: SizingMode,
+) -> SizeAndBaselines {
+    compute(tree, node, known_dimensions, parent_size, available_space, sizing_mode)
+}
 
-    fn measure_size(
-        tree: &mut impl LayoutTree,
-        node: NodeId,
-        known_dimensions: Size<Option<f32>>,
-        parent_size: Size<Option<f32>>,
-        available_space: Size<AvailableSpace>,
-        sizing_mode: SizingMode,
-    ) -> Size<f32> {
-        compute(tree, node, known_dimensions, parent_size, available_space, sizing_mode).size
-    }
+/// Measure a leaf node's size
+pub(crate) fn measure_size(
+    tree: &mut Taffy,
+    node: NodeId,
+    known_dimensions: Size<Option<f32>>,
+    parent_size: Size<Option<f32>>,
+    available_space: Size<AvailableSpace>,
+    sizing_mode: SizingMode,
+) -> Size<f32> {
+    compute(tree, node, known_dimensions, parent_size, available_space, sizing_mode).size
 }
 
 /// Compute the size of a leaf node (node with no children)
 pub(crate) fn compute(
-    tree: &mut impl LayoutTree,
+    tree: &mut Taffy,
```

Comment:
Why are these things using Taffy again instead of the impl Trait?
I thought that's what made the approach more generic

Reply:
Nvm, I should have reviewed the other PR first :D

Reply:
Because it allows us to remove 3 methods from the `LayoutTree` trait, and these functions are private / only ever used with `Taffy` anyway:

- I don't think it makes sense to expose the `leaf` functions because they are specific to measure_funcs which are specific to the `Taffy` struct.
- I don't think it makes sense to expose the "generic" functions as anyone implementing the `LayoutTree` trait would likely want to create their own version of these functions anyway (including the abiity to switch into their own internal layout methods such as text layout).

See #431.